### PR TITLE
fix(python): drop simple-rlp dependency and use internal copy

### PR DESCRIPTION
**Pull request by prusnak**
supersedes https://github.com/trezor/trezor-firmware/pull/3046

not tested, please test before merging

### CODE REVIEW

**Comment by matejcik** [comment]
simplified to just the encode method, and further simplified using code that we already have in core :)
added some tests

File: python/src/trezorlib/_rlp.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def _byte_size(x: int) -> int:
+    if x < 0:
+        raise ValueError("only unsigned ints are supported")
+    return (x.bit_length() + 7) // 8
+
+
+def _int_to_bytes(n: int) -> bytes:
+    """Convert python objects to bytes can be deserialize with converters"""
```
Comment (mmilata) [inline python/src/trezorlib/_rlp.py:17]: nit: kinda hard to parse

File: python/tests/test_rlp.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    (b"A" * 1024, b"\xb9\x04\x00" + b"A" * 1024),
+    ([b"dog", b"cat", [b"spy"]], b"\xcd\x83dog\x83cat\xc4\x83spy"),
+    ([b"A" * 1024], b"\xf9\x04\x03\xb9\x04\x00" + b"A" * 1024),
+    ([], b"\xc0"),
+    (b"", b"\x80"),
+    (1, b"\x01"),
+    (0x7F, b"\x7f"),
+    (0x80, b"\x81\x80"),
+    (0x1_0000_0001, b"\x85\x01\x00\x00\x00\x01"),
+    ([0x1234, 0x5678], b"\xc6\x82\x12\x34\x82\x56\x78"),
```
Comment (mmilata) [inline python/tests/test_rlp.py:17]: nit: testcase for the `len > 55` branch would be nice

**Review by mmilata — APPROVED** [review]

File: python/src/trezorlib/_rlp.py (hunk 1 above)
Comment (matejcik) [reply python/src/trezorlib/_rlp.py]: yeah that would be because it comes from a different function
eba3577832a788ab7cc56008551410f88b71224b

File: python/tests/test_rlp.py (hunk 2 above)
Comment (matejcik) [reply python/tests/test_rlp.py:25]: eba3577832a788ab7cc56008551410f88b71224b

**Review by mmilata — APPROVED** [review]
